File: app/src/main/java/app/com/ezata/adapter/OrderDetailAdapter.java
package app.com.ezata.adapter;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import app.com.ezata.R;
import app.com.ezata.model.NewOrder;
import app.com.ezata.model.OrderDetail;

public class OrderDetailAdapter extends RecyclerView.Adapter<OrderDetailAdapter.ViewHolder> {
    OrderDetail[] orderList;

    public OrderDetailAdapter(OrderDetail[] orderList) {
        this.orderList = orderList;
    }

    @NonNull
    @Override
    public ViewHolder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        LayoutInflater layoutInflater = LayoutInflater.from(parent.getContext());
        View listItem = layoutInflater.inflate(R.layout.order_detail_item, parent, false);
        OrderDetailAdapter.ViewHolder viewHolder = new OrderDetailAdapter.ViewHolder(listItem);
        return viewHolder;
    }

    @Override
    public void onBindViewHolder(@NonNull ViewHolder holder, int position) {
        OrderDetail orderDetail = orderList[position];
        holder.tvQty.setText(orderDetail.getQuantity());
        holder.tvItem.setText(orderDetail.getItems());
        holder.tvPrice.setText(orderDetail.getPrice());
        if (orderDetail.getInstruction().isEmpty()) {
            holder.tvInstruction.setVisibility(View.GONE);
        } else {
            holder.tvInstruction.setText(orderDetail.getInstruction());
        }    }

    @Override
    public int getItemCount() {
        return orderList.length;
    }

    public class ViewHolder extends RecyclerView.ViewHolder {
        TextView tvQty, tvItem, tvPrice,tvInstruction;

        public ViewHolder(@NonNull View itemView) {
            super(itemView);
            tvQty = itemView.findViewById(R.id.tvQty);
            tvItem = itemView.findViewById(R.id.tvItem);
            tvPrice = itemView.findViewById(R.id.tvPrice);
            tvInstruction=itemView.findViewById(R.id.tvInstruction);

        }
    }
}

File: app/src/main/java/app/com/ezata/adapter/OrderDetailAdapter.kt
package app.com.ezata.adapter

import app.com.ezata.model.OrderDetail
import androidx.recyclerview.widget.RecyclerView
import android.view.ViewGroup
import android.view.LayoutInflater
import android.view.View
import app.com.ezata.R
import android.widget.TextView
import app.com.ezata.model.OrderItems

class OrderDetailAdapter() :
    RecyclerView.Adapter<OrderDetailAdapter.ViewHolder>() {


    var orderItems = arrayListOf<OrderItems>()
        set(value) {
            field = value
            notifyDataSetChanged()
        }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): ViewHolder {
        val layoutInflater = LayoutInflater.from(parent.context)
        val listItem =
            layoutInflater.inflate(R.layout.order_detail_item, parent, false)
        return ViewHolder(listItem)
    }

    override fun onBindViewHolder(holder: ViewHolder, position: Int) {
        val orderDetail = orderItems[position]
        holder.tvQty.text = "${orderDetail.ordersMenuParse?.quantity}x"
        holder.tvItem.text = orderDetail.ordersMenuParse?.menu?.dishTitle
        holder.tvPrice.text = "$${orderDetail.ordersMenuParse?.price}"

        val extraItem = orderDetail.ordersMenuExtraItemParse.joinToString(separator = "") {
            "${it?.menuExtraItem?.name} $${it?.menuExtraItem?.price}\n"
        }

        if(extraItem.isNotBlank()){
            holder.tvExtraItem.text = extraItem
            holder.tvExtraItem.visibility = View.VISIBLE
        }else{
            holder.tvExtraItem.visibility = View.GONE
        }

    }

    override fun getItemCount(): Int {
        return orderItems.size
    }

    inner class ViewHolder(itemView: View) : RecyclerView.ViewHolder(itemView) {
        var tvQty: TextView = itemView.findViewById(R.id.tvQty)
        var tvItem: TextView = itemView.findViewById(R.id.tvItem)
        var tvPrice: TextView = itemView.findViewById(R.id.tvPrice)
        var tvInstruction: TextView = itemView.findViewById(R.id.tvInstruction)
        var tvExtraItem: TextView = itemView.findViewById(R.id.extraItemTv)

    }
}